tap_github/repository_streams.py

```diff
@@ -1307,3 +1307,55 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class StatsContributorsStream(GitHubStream):
+    """
+    Defines 'StatsContributors' stream. Fetching contributors activity.
+    https://docs.github.com/en/rest/reference/metrics#get-all-contributor-commit-activity
+    """
+
+    name = "stats_contributors"
+    path = "/repos/{org}/{repo}/stats/contributors"
+    primary_keys = ["org", "repo", "user_id"]
+    parent_stream_type = RepositoryStream
+    ignore_parent_replication_key = False
+    state_partitioning_keys = ["repo", "org"]
+    # Note - these queries are expensive and the API might return an HTTP 202 if the response
+    # has not been cached recently. https://docs.github.com/en/rest/reference/metrics#a-word-about-caching
+    tolerated_http_errors = [202]
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        """Parse the response and return an iterator of flattened contributor activity."""
+        parsed_response = super().parse_response(response)
+        for contributor_activity in parsed_response:
+            weekly_data = contributor_activity["weeks"]
+            for week in weekly_data:
+                # no need to save weeks with no contributions.
+                is_week_empty = sum(week[key] for key in ["a", "c", "d"]) > 0
```

Comment:
How do we distinguish a missing response (say if we received a 202 that has not been retried yet, something like `undefined`) from this situation where it's a `zero`, in the downstream processing?

Reply:
I don't think that's relevant, we get data for the past year each time, so ultimately we will get the data we want. The only issue is if we ALWAYS get a 202...